six/modules/c++/six/build/makeEnums.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+
+#
+# =========================================================================
+# This file is part of six-c++
+# =========================================================================
+#
+# (C) Copyright 2004 - 2015, MDA Information Systems LLC
+#
+# six.sicd-python is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; If not,
+# see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import subprocess
+
+script = os.path.join(os.getcwd(), 'externals', 'coda-oss',
+                      'build', 'scripts', 'makeEnums.py')
+
+inputFile = os.path.join(os.path.split(__file__)[0], 'enums.txt')
+outDir = os.path.join(os.path.split(__file__)[0], os.pardir, 'include', 'six')
+
+assert os.path.exists(script)
+assert os.path.exists(inputFile)
+assert os.path.exists(outDir)
+
+target = os.path.join(outDir, 'Enums.h')
+if os.path.exists(target):
+    os.remove(target)
+    
+with open(target, 'w') as f:
+    f.write("""/* =========================================================================
```

Comment:
Fixed.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Tuesday, April 05, 2016 10:50 AM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>
Subject: Re: [ngageoint/six-library] Add script to call and process output of makeEnums.py in CODA-OSS (#59)


In six/modules/c++/six/build/makeEnums.py<http://cp.mcafee.com/d/5fHCN0pdEIfCzBVMsed7apKVJBxAsed7bXVJYsOMOyqejhOrhKZtVMQsTd7bNJ55B5dYQsKcfCN6IdCjS7a_Ia0Fl3mWsKrlgRKDbCTnHEcT7fZvCnNRQS1TnKnjpsd7feecfK8Ith5dqWqJTCel3PWApmU6CQjqpK_8K6zBV55BeXNKVIDeqR4IMzapdIxO-g8Wh-IEjOAwYKv8x7oBImrffNYQswoxaBCWkBm0H5ZVZZ5ZYsrxmfH7BwkhBcCSgVsS-MYUr1vF6y0QJM8Wh-IEjOAwYKv8x7oBziWq81110QgkJ6iQPh1eFEwJFVVEwS21EwmzmTQEq8aJPmrCq8aKCy1niLzk6RqsImd44E6y0i30s85MMc31Ewd506fCzAQkCPhOqejt0e2>:

> +script = os.path.join(os.getcwd(), 'externals', 'coda-oss',

> +                      'build', 'scripts', 'makeEnums.py')

> +

> +inputFile = os.path.join(os.path.split(__file__)[0], 'enums.txt')

> +outDir = os.path.join(os.path.split(__file__)[0], os.pardir, 'include', 'six')

> +

> +assert os.path.exists(script)

> +assert os.path.exists(inputFile)

> +assert os.path.exists(outDir)

> +

> +target = os.path.join(outDir, 'Enums.h')

> +if os.path.exists(target):

> +    os.remove(target)

> +

> +with open(target, 'w') as f:

> +    f.write("""/* =========================================================================

Can't you just call the coda-oss makeEnums.py and point it to six/build/Enums.h.template with the -t flag?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/1jWVIp40USyM-qenD1MUQsFCXCSm6hMUQsLLCTNPb3a9EVd79J6XRTD3hPsQsL6QkmkkTPhOUM-r4qMSpfosH-ME2BkdrFOVJl3mWsKrtuKwPss_R-pv7njo7tuVtdBMQsYUUM-UyNR4kRHFGTuoVkffGhBrwqrjdFCXYyUqenAkmkXL6XCOsVHkiP2cFASO7bV0zF7WOxfai3OVYy4tymNpIY_7PhPzIz_FDKfFSoohgZddVxZCNEVtd7bO1EVpogKghR5_tvgvjjQ38LLfLELLzzsaNZoYI2ycFASO7bCTS7D3obZ8Qg6BK17ifRB2ukA7BPV48X4Iqnjh08886y2BEOmCq89Rd45Jffd46Mgd42QqS-B3h1lKqPsPh1lQQgaWlYqwSHjByNEwB0Qg2go3x0K61wod41EE0NYQsCyASqejhOrz5oYaCixCnJ6>


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.